python/ray/autoscaler/_private/kuberay/autoscaling_config.py

```diff
@@ -280,46 +277,53 @@ def _get_num_cpus(
 def _get_memory(
     ray_start_params: Dict[str, str], k8s_resource_limits: Dict[str, Any]
 ) -> Optional[int]:
-    """Get memory resource annotation from ray_start_params, if it is set there.
-
-    TODO, maybe: Consider container resource limits as in
-    https://github.com/ray-project/ray/pull/14567/files
+    """Get memory resource annotation from ray_start_params or k8s_resource_limits,
+    with priority for ray_start_params.
     """
     if "memory" in ray_start_params:
         return int(ray_start_params["memory"])
+    elif "memory" in k8s_resource_limits:
+        memory_quantity: str = k8s_resource_limits["memory"]
+        return _round_up_k8s_quantity(memory_quantity)
     return None
 
 
+def _round_up_k8s_quantity(quantity: str) -> int:
+    """Rounds a Kubernetes resource quantity up to the nearest integer.
+
+    Args:
+        quantity: Resource quantity as a string in the canonical K8s form.
+
+    Returns:
+        The quantity, rounded up, as an integer.
+    """
+    resource_decimal: decimal.Decimal = kubernetes.utils.quantity.parse_quantity(
+        quantity
+    )
+    rounded = resource_decimal.to_integral_value(rounding=decimal.ROUND_UP)
+    return int(rounded)
+
+
 def _get_num_gpus(
     ray_start_params: Dict[str, str],
     k8s_resource_limits: Dict[str, Any],
     group_name: str,
 ) -> Optional[int]:
-    """Read the number of GPUs from the Ray start params.
-
-    Potential TODO: Read GPU info from the container spec, here and in the
-    Ray Operator.
+    """Get memory resource annotation from ray_start_params or k8s_resource_limits,
+    with priority for ray_start_params.
     """
 
     if "num-gpus" in ray_start_params:
         return int(ray_start_params["num-gpus"])
-
-    # Issue a warning if GPUs are present in the container spec but not in the
-    # ray start params.
-    # TODO: Consider reading GPU info from container spec.
-    for key in k8s_resource_limits:
-        global _GPU_WARNING_LOGGED
-        if "gpu" in key and not _GPU_WARNING_LOGGED:
-            with suppress(Exception):
-                if int(k8s_resource_limits[key]) > 0:
-                    logger.warning(
-                        f"Detected GPUs in container resources for group {group_name}."
-                        "To ensure Ray and the autoscaler are aware of the GPUs,"
-                        " set the `--num-gpus` rayStartParam."
-                    )
-                    _GPU_WARNING_LOGGED = True
-            break
-
+    else:
+        for key in k8s_resource_limits:
+            # e.g. nvidia.com/gpu
+            if key.endswith("/gpu"):
+                num_gpus = int(k8s_resource_limits[key])
```

Comment:
Is this always going to be an int? In the other cases it is a string and you call _round_up_k8s_quantity

Reply:
Good point
It's probably always going to be an int represented as a string.
However, to be extra safe I've switched to using `_round_up_k8s_quantity` for this as well.